UiComponent2_Transition.md

```diff
@@ -0,0 +1,120 @@
+# UiComponent2 Transition Notes
+> A document to explain the new features and breakages when converting components from Component / UiComponent to 
+Component2 / UiComponent2.
+---
+* __[What's New in Component2](#whats-new-in-component2)__
+* __[Why a new base class?](#why-a-new-base-class)__
+* __[New Features, improvements, and breakages](#new-features-improvements-and-breakages)__
+    * [Small Perf Improvements](#small-perf-improvements)
+    * [JS-Backed Maps](#js-backed-maps)
+    * [Lifecycle Method Changes](#lifecycle-method-changes)
+    * [Other Method Changes](#other-method-changes)
+    * [PropTypes](#prop-validation-goes-through-proptypes)
+    * [Other Breaking Changes](#other-breakages-we-made-while-we-had-the-opportunity)
+* __[Updating](#updating)__
+## What's New in Component2
+`Component` and `UiComponent` are being deprecated in favor of new versions of the API: `Component2` and `UiComponent2`.
+
+#### tl;dr
+
+- Pros: slightly faster, improved dev experience, easier to maintain, easier integration with JS libs
+- Cons: breaks a few advanced component APIs that will require conversion, some of which will be automated with codemod
+
+Migration can happen at any time, and is opt-in. We'll provide a codemod to handle the simpler cases and run it as 
```

Comment:
Should we / do we have a section that explains how to run the codemod?

Reply:
I have the update section at the end - I added a link on the word "codemod" to take them there to make it a little more clear. I could also see moving that section to the top if that's not clear enough though.